discord_bot/backfill.py

```
@@ -57,10 +57,19 @@ async def backfill_channel(channel, target_limit: int = CONTEXT_AGENT_MAX_MESSAG
             else:
                 # No data, full fetch
                 logger.info(f"[Backfill] ⚡ No existing data for {channel_name}. Performing initial fetch...")
-                fetched_count = len(await fetch_and_cache_from_api(channel, limit=target_limit))
+                try:
+                    initial_messages = await fetch_and_cache_from_api(channel, limit=target_limit)
+                    fetched_count = len(initial_messages)
+                    logger.info(f"[Backfill] Initial fetch for {channel_name} returned {fetched_count} messages")
```

Ruff (0.15.2)

[warning] Logging statement uses f-string

(G004)